packages/Php74/src/Rector/Property/TypedPropertyRector.php

```diff
@@ -103,8 +101,32 @@ public function refactor(Node $node): ?Node
             return null;
         }
 
+        $this->removeVarPhpTagValueNodeIfNotComment($node);
+
         $node->type = $propertyTypeNode;
 
         return $node;
     }
+
+    private function removeVarPhpTagValueNodeIfNotComment(Property $property): void
+    {
+        $propertyPhpDocInfo = $this->getPhpDocInfo($property);
+        // nothing to remove
+        if ($propertyPhpDocInfo === null) {
+            return;
+        }
+
+        $varTagValueNode = $propertyPhpDocInfo->getByType(VarTagValueNode::class);
+        if ($varTagValueNode === null) {
+            return;
+        }
+
+        // has description? keep it
+        if ($varTagValueNode->description !== '') {
+            return;
+        }
+
+        $propertyPhpDocInfo->removeByType(VarTagValueNode::class);
```

Comment:
I don't see a condition that would preserve useful annotations such as

```
/**
 * @var array<int, string>
 */
private array $foo;
```

Is there a test that annotations with generic types are NOT removed?

Reply:
I need to add it

Reply:
Added